spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -453,4 +467,21 @@ private static void commitChanges(Table table, SetProperty setLocation, SetPrope
 
     transaction.commitTransaction();
   }
+
+  private static boolean isPathIdentifier(Identifier ident) {
+    return ident instanceof PathIdentifier;
+  }
+
+  private static void checkNotPathIdentifier(Identifier identifier, String method) {
+    if (identifier instanceof PathIdentifier) {
+      throw new IllegalArgumentException(String.format("Cannot pass path based identifier to %s method. %s is a path.",
+          method, identifier));
+    }
+  }
+
+  private <T> T pathOrTable(Identifier ident, Function<String, T> path, Function<TableIdentifier, T> table) {
```

Comment:
Looks like there are 3 ways this is called:
1. To load a table
2. To create a builder
3. To drop a table

I think it would be a bit cleaner to make the first 2 into private methods (`load` and `newBuilder`) rather than using `pathOrTable` with the same functions in lots of places.